PurchasesTests/Purchasing/ProductInfoTests.swift

```diff
@@ -0,0 +1,273 @@
+import XCTest
+import Nimble
+
+import Purchases
+
+class ProductInfoTests: XCTestCase {
+    func testAsDictionaryConvertsProductIdentifierCorrectly() {
+        let product_identifier = "cool_product"
+        let productInfo = RCProductInfo(productIdentifier: product_identifier,
+                                        paymentMode: .none,
+                                        currencyCode: "UYU",
+                                        price: 9.99,
+                                        normalDuration: nil,
+                                        introDuration: nil,
+                                        introDurationType: .none,
+                                        introPrice: nil,
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
+        expect(productInfo.asDictionary()["product_id"] as? String) == product_identifier
+    }
+
+    func testAsDictionaryConvertsPaymentModeCorrectly() {
+        var paymentMode: RCPaymentMode = .none
+        var productInfo = RCProductInfo(productIdentifier: "cool_product",
+                                        paymentMode: paymentMode,
+                                        currencyCode: "UYU",
+                                        price: 9.99,
+                                        normalDuration: nil,
+                                        introDuration: nil,
+                                        introDurationType: .none,
+                                        introPrice: nil,
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
```

Comment:
In this test you could create the empty product info I mentioned above and then just modify the paymentMode to cover all cases